File: python-wrapper/src/python/model/embed/PatientHealthCarePartyType.py
from typing import Union, Dict
from enum import Enum

class PatientHealthCarePartyType(Enum):
	Doctor = "doctor"
	Referral = "referral"
	Medicalhouse = "medicalhouse"
	Retirementhome = "retirementhome"
	Hospital = "hospital"
	Other = "other"
	Referringphysician = "referringphysician"
	Managingorganization = "managingorganization"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'PatientHealthCarePartyType':
		if data == "doctor":
			return PatientHealthCarePartyType.Doctor
		elif data == "referral":
			return PatientHealthCarePartyType.Referral
		elif data == "medicalhouse":
			return PatientHealthCarePartyType.Medicalhouse
		elif data == "retirementhome":
			return PatientHealthCarePartyType.Retirementhome
		elif data == "hospital":
			return PatientHealthCarePartyType.Hospital
		elif data == "other":
			return PatientHealthCarePartyType.Other
		elif data == "referringphysician":
			return PatientHealthCarePartyType.Referringphysician
		elif data == "managingorganization":
			return PatientHealthCarePartyType.Managingorganization
		else:
			raise Exception(f"{data} is not a valid value for PatientHealthCarePartyType enum.")

__all__ = ['PatientHealthCarePartyType']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/embed/PatientHealthCarePartyType.kt
package com.icure.cardinal.sdk.model.embed

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class PatientHealthCarePartyType(
	internal val dtoSerialName: String,
) {
	@SerialName("doctor")
	Doctor("doctor"),

	@SerialName("referral")
	Referral("referral"),

	@SerialName("medicalhouse")
	Medicalhouse("medicalhouse"),

	@SerialName("retirementhome")
	Retirementhome("retirementhome"),

	@SerialName("hospital")
	Hospital("hospital"),

	@SerialName("other")
	Other("other"),

	@SerialName("referringphysician")
	Referringphysician("referringphysician"),

	@SerialName("managingorganization")
	Managingorganization("managingorganization"),
}
